daisy/workflow/step.go

```diff
@@ -0,0 +1,154 @@
+package workflow
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+)
+
+type stepImpl interface {
+	run(s *Step) error
+	validate(s *Step) error
+}
+
+// Step is a single daisy workflow step.
+type Step struct {
+	name string
+	w    *Workflow
```

Comment:
Is this the only field in Step that a stepImpl needs? If so would it make more sense to put this into step that way you don't have to pass it into stepImpl.run and stepImpl.validate?